File: vue.jsonforms/src/entities/owners-and-controllers/IndividualController.ts
import { IndividualControllerDetails } from "./IndividualControllerDetails";
import { FinancialStatus } from "./FinancialStatus";
import { CompanyOfficerAppointmentDetails } from "./CompanyOfficerAppointmentDetails";
import { Controller } from "@/entities/firm-details/Controller";
import { v4 as uuid } from "uuid";

export class IndividualController {
  public id: string = uuid();
  public detail: IndividualControllerDetails =
    new IndividualControllerDetails();
  public directorsAndDirectorship: CompanyOfficerAppointmentDetails[] =
    [] as CompanyOfficerAppointmentDetails[];
  public controllingInterests: Controller[] = [] as Controller[];
  public financialStatus: FinancialStatus = new FinancialStatus();
  public curriculumVitaeUrls: string[] = [];
}

File: CSharp.Functions/Common/Entities/FirmDetails/OwnersAndControllers/IndividualController.cs
namespace Common.Entities
{
    public class IndividualController
    {
        public IndividualControllerDetails? Detail { get; set; }
        public List<CompanyOfficerAppointmentDetails> DirectorsAndDirectorship { get; set; } = new();
        public List<CompanyControllingInterestDetails> ControllingInterests { get; set; } = new();
        public FinancialStatus? FinancialStatus { get; set; }
        public List<string> CurriculumVitaeUrls { get; set; } = new();
    }
}